distro/run/core/src/test/java/org/camunda/bpm/run/test/plugins/CamundaRunProcessEnginePluginsRegistrationTest.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = { CamundaBpmRun.class })
+@ActiveProfiles(profiles = { "test-new-plugins" }, inheritProfiles = true)
+public class CamundaRunProcessEnginePluginsRegistrationTest {
+
+  @Autowired
+  protected ProcessEngineConfigurationImpl processEngineConfiguration;
+
+  @Autowired
+  protected CamundaBpmRunProperties properties;
+
+  protected List<ProcessEnginePlugin> plugins;
+
+  @Before
+  public void setUp() {
+    this.plugins = processEngineConfiguration.getProcessEnginePlugins();
+  }
+
+  @Test
+  public void shouldPickUpAllPluginConfigurations() {
+    // given a CamundaBpmRunProperties instance
+    String pluginOne = "org.camunda.bpm.run.test.plugins.TestFirstPlugin";
+    String pluginTwo = "org.camunda.bpm.run.test.plugins.TestSecondPlugin";
+
+    // then
+    // assert that any plugin configuration properties were mapped properly
+    assertThat(properties.getProcessEnginePlugins()).hasSize(2);
+    assertThat(properties.getProcessEnginePlugins().keySet())
+        .contains(pluginOne, pluginTwo);
+
+    Map<String, Object> firstPluginMap = properties.getProcessEnginePlugins().get(pluginOne);
+    assertThat(firstPluginMap).hasSize(2);
+    assertThat(firstPluginMap.keySet()).contains("parameterOne", "parameterTwo");
+    assertThat(firstPluginMap.values()).contains("valueOne", true);
+
+    Map<String, Object> secondPluginMap = properties.getProcessEnginePlugins().get(pluginTwo);
+    assertThat(secondPluginMap).hasSize(3);
+    assertThat(secondPluginMap.keySet())
+        .contains("parameterOne", "parameterTwo", "parameterThree");
+    assertThat(secondPluginMap.values()).contains(1.222, false, 123);
+  }
+
+  @Test
+  public void shouldRegisterYamlDefinedPluginsWithProcessEngine() {
+    // given a yaml config file defining process engine plugins
+
+    // then
+    // there is a single composite plugin
+    assertThat(plugins).hasSize(1).hasOnlyElementsOfType(CompositeProcessEnginePlugin.class);
+    CompositeProcessEnginePlugin compositePlugin =
+        (CompositeProcessEnginePlugin) plugins.get(0);
+
+    // the composite plugin contains all of the registered plugins
+    List<ProcessEnginePlugin> registeredPlugins = compositePlugin.getPlugins();
+    List<Class> classList = registeredPlugins.stream().map(ProcessEnginePlugin::getClass)
+        .collect(Collectors.toList());
+    assertThat(classList).contains(TestFirstPlugin.class, TestSecondPlugin.class);
+  }
+
+  @Test
+  public void shouldInitializeRegisteredPlugins() {
+    // given
+    assertThat(plugins).hasSize(1).hasOnlyElementsOfType(CompositeProcessEnginePlugin.class);
```

Comment:
```suggestion
```

Assertions in the given block are against our style guide. We have a test case already that covers that a composite process engine plugin is used (shouldRegisterYamlDefinedPluginsWithProcessEngine). I think we don't need to re-test it.